### PR TITLE
HO-57 MANTAS TESTING BRANCH - ALL MERGED CODE

### PR DESCRIPTION
**Pull request by mantaslv**
merging into yours thuy because i want you to see what i did in a combined branch, think ive cleaned up the code fully and everything works properly for all users, so yeah pull into it and check it out and see if it does everything you managed to get yours to do!

### CODE REVIEW

File: frontend/src/components/CreateForm.js [hunk 1]
```diff
@@ -78,7 +72,7 @@ const CreateForm = (props) => {
             </select>
             <label htmlFor="firstLine">Address</label>
             <input type="text" name="firstLine" id="firstLine" />
-            <label htmlFor="city">City</label>
+            <label htmlFor="city">Town</label>
```
Comment (mantaslv) [inline frontend/src/components/CreateForm.js:75]: changed it to town cos it a bit more specific if everything is in london

File: frontend/src/pages/CommentForm.js [hunk 2]
```diff
@@ -4,62 +4,55 @@ import { useComment } from "../hooks/useComment";
...
+            if (response.ok) {
+                if (userType === 'org') setUserName(json.organisationName);
+                if (userType === 'ind') setUserName(json.firstName + " " + json.lastName);
+            } else {
+                dispatch({ type: "SET_ERROR", payload: json });
+            }
+        };
+
+        getUserName();
+    }, [userName]);
```
Comment (mantaslv) [inline frontend/src/pages/CommentForm.js:30]: bit of trial and error and found this was a much simpler way of setting the username and fetching with a second template literal

File: frontend/src/pages/ListingsFeed.js [hunk 3]
```diff
@@ -27,40 +30,61 @@ const Listings = () => {
...
+    }, [dispatch, isUpdated]);
 
     let listingsToShow;
     switch (listingRequirement) {
         case 'volunteering':
             listingsToShow = listings.filter(listing => listing.requirement === 'Volunteering')
             break;
         case 'donation':
-            listingsToShow = listings.filter(listing => listing.requirement === 'Donation of Goods')
+            listingsToShow = listings.filter(listing => listing.requirement === 'Donation of goods')
```
Comment (mantaslv) [inline frontend/src/pages/ListingsFeed.js:87]: this fixed the filter

File: frontend/src/components/CreateForm.js [hunk 4]
```diff
@@ -61,14 +58,13 @@ const CreateForm = (props) => {
         }
         // props is passed in from ListingsFeed.js
         props.createListing(listing);
+        setShowSubmittedMessage(true);
```
Comment (mantaslv) [inline frontend/src/components/CreateForm.js:61]: thanks thuy works perfect

File: frontend/src/pages/CommentForm.js (hunk 2 above)
Comment (mantaslv) [reply frontend/src/pages/CommentForm.js:30]: yeah in hindsight maybe user name would have made more sense at the start of the project, my bad! we'll implement it in the next iteration ;)

**Review by tlchambers — APPROVED** [review]

**Review by AndreeaDinculescu — APPROVED** [review]

**Pull request merged**